lib/events.js

```diff
@@ -306,10 +306,25 @@ EventEmitter.prototype.prependListener =
     };
 
 function onceWrapper() {
-  this.target.removeListener(this.type, this.wrapFn);
   if (!this.fired) {
+    this.target.removeListener(this.type, this.wrapFn);
     this.fired = true;
-    this.listener.apply(this.target, arguments);
+    switch (arguments.length) {
+      case 0:
+        return this.listener.call(this.target);
+      case 1:
+        return this.listener.call(this.target, arguments[0]);
+      case 2:
+        return this.listener.call(this.target, arguments[0], arguments[1]);
+      case 3:
+        return this.listener.call(this.target, arguments[0], arguments[1],
+                                  arguments[2]);
+      default:
+        const args = new Array(arguments.length);
```

Comment:
No, because you cannot pass `arguments` without penalty except in only one or two cases. `apply()` is supposed to be one of those cases (which is what was used here before) but for some reason it can still cause a deopt (noticed this in some `net` benchmarks and tests).